statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/feedback/FeedbackUnionOperator.java

```diff
@@ -43,20 +48,21 @@
 
   // -- configuration
   private final FeedbackKey<T> feedbackKey;
-  private final SerializablePredicate<T> isBarrierMessage;
+  private final SerializableFunction<T, OptionalLong> isBarrierMessage;
   private final SerializableFunction<T, ?> keySelector;
   private final long totalMemoryUsedForFeedbackCheckpointing;
   private final TypeSerializer<T> elementSerializer;
 
   // -- runtime
-  private transient UnboundedFeedbackLogger<T> feedbackLogger;
+  private transient UnboundedFeedbackLoggerFactory<T> feedbackLoggerFactory;
+  private transient Map<Long, UnboundedFeedbackLogger<T>> uncompletedCheckpoints;
   private transient boolean closedOrDisposed;
   private transient MailboxExecutor mailboxExecutor;
   private transient StreamRecord<T> reusable;
 
   FeedbackUnionOperator(
```

Comment:
Perhaps a bit of a nitpick, but with the added functionality to this class, we're now slightly lacking in unit test coverage.

Before having just the `UnboundedFeedbackLoggerTest` is sufficient as it already essentially encapsulates the core logic.

Maybe moving some of the new methods, such as `commitCheckpoint(maybeCheckpoint.getAsLong());` / `appendToCheckpoints(element);` and the `uncompletedCheckpoints` map, essentially the logic of bookkeeping multiple loggers, to a separate `UnboundedFeedbackLoggers` class, and have a unit test for that is enough.